<a href="https://colab.research.google.com/github/shambhavidpsg-beep/Shambhavi4C2K24CSUN01200/blob/main/ML_Lab_3.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
import pandas as pd

In [2]:
df= pd.read_csv('Insuramce Ml LAB.csv')

Deleting some values in insurance data and dropping the missing/null values

In [4]:
df.isnull( )

age    sex    bmi  children  smoker  region  charges
0     False  False  False     False   False   False    False
1     False  False  False     False   False   False    False
2     False  False  False     False   False   False    False
3     False  False  False     False   False   False    False
4     False  False  False     False   False   False    False
...     ...    ...    ...       ...     ...     ...      ...
1333  False  False  False     False   False   False    False
1334  False  False  False     False   False   False    False
1335  False  False  False     False   False   False    False
1336  False  False  False     False   False   False    False
1337  False  False  False     False   False   False    False

[1338 rows x 7 columns]

In [5]:
df.isnull().sum()

age         0
sex         0
bmi         4
children    1
smoker      2
region      1
charges     1
dtype: int64

Shape is run because it tells the size of the data i.e rowxcolumn whihc is 1338 tells rows and 7 tells about columns

In [7]:
df.shape

(1338, 7)

We delete the missing values

In [8]:
df= df.dropna()

In [18]:
#Removing 4th indexed value from the dataframe
df.drop(4).head()

age     sex     bmi  children smoker     region      charges
0   19  female  27.900       0.0    yes  southwest  16884.92400
1   18    male  33.770       1.0     no  southeast   1725.55230
2   28    male  33.000       3.0     no  southeast   4449.46200
3   33    male  22.705       0.0     no  northwest  21984.47061
5   31  female  25.740       0.0     no  southeast   3756.62160

In [19]:
df.rename({0:"First",1:"Second"})

age     sex     bmi  children smoker     region      charges
First    19  female  27.900       0.0    yes  southwest  16884.92400
Second   18    male  33.770       1.0     no  southeast   1725.55230
2        28    male  33.000       3.0     no  southeast   4449.46200
3        33    male  22.705       0.0     no  northwest  21984.47061
4        32    male  28.880       0.0     no  northwest   3866.85520
...     ...     ...     ...       ...    ...        ...          ...
1333     50    male  30.970       3.0     no  northwest  10600.54830
1334     18  female  31.920       0.0     no  northeast   2205.98080
1335     18  female  36.850       0.0     no  southeast   1629.83350
1336     21  female  25.800       0.0     no  southwest   2007.94500
1337     61  female  29.070       0.0    yes  northwest  29141.36030

[1330 rows x 7 columns]

In [20]:
#Creates a new column with all the values equal to 1
df['NewColumn'] = 1
df.head()

age     sex     bmi  children smoker     region      charges  NewColumn
0   19  female  27.900       0.0    yes  southwest  16884.92400          1
1   18    male  33.770       1.0     no  southeast   1725.55230          1
2   28    male  33.000       3.0     no  southeast   4449.46200          1
3   33    male  22.705       0.0     no  northwest  21984.47061          1
4   32    male  28.880       0.0     no  northwest   3866.85520          1

In [22]:
df.sort_values(by='bmi', ascending=False).head()

age     sex    bmi  children smoker     region     charges  NewColumn
1317   18    male  53.13       0.0     no  southeast   1163.4627          1
1047   22    male  52.58       1.0    yes  southeast  44501.3982          1
847    23    male  50.38       1.0     no  southeast   2438.0552          1
116    58    male  49.06       0.0     no  southeast  11381.3254          1
286    46  female  48.07       2.0     no  northeast   9432.9253          1

In [23]:
df.sort_values(by=['bmi','charges']).head(10)

age     sex     bmi  children smoker     region      charges  NewColumn
172    18    male  15.960       0.0     no  northeast   1694.79640          1
428    21  female  16.815       1.0     no  northeast   3167.45585          1
1226   38    male  16.815       2.0     no  northeast   6640.54485          1
412    26  female  17.195       2.0    yes  northeast  14455.64405          1
1286   28  female  17.290       0.0     no  northeast   3732.62510          1
1029   37  female  17.290       2.0     no  northeast   6877.98010          1
250    18    male  17.290       2.0    yes  northeast  12829.45510          1
28     23    male  17.385       1.0     no  northwest   2775.19215          1
680    21  female  17.400       1.0     no  southwest   2585.26900          1
410    19    male  17.480       0.0     no  northwest   1621.34020          1

In [24]:
#Creating dataframe1
df1 = pd.DataFrame({
        'children':[0.0, 2.0, 3.0, 5.0],
        'age':[25, 24, 52, 40],
        'smoker':['yes', 'yes', 'no', 'no']})
df1

children  age smoker
0       0.0   25    yes
1       2.0   24    yes
2       3.0   52     no
3       5.0   40     no

In [27]:
# Apply function
def fun(value):
    if value > 70:
        return "Yes"
    else:
        return "No"

df['age'] = df['age(1-30)'].apply(fun)
df.head(10)

KeyError: 'age(1-30)'